### DISCUSSION TITLE
use type given in argument as return type (without inference)

### DISCUSSION THREAD

**Discussion by karolzlot** [microsoft/pyright#5390] [Q&A] [answered]
Related issue: https://github.com/microsoft/pyright/issues/5386

Is it possible to use overload and TypeVar to achieve something like this? Or maybe some other solution exists?

In issue above I found some ways to achieve this with use of pyright type inference, but I wonder if some other ways (without inference) are possible.
What other mechanisms could be used?
Does it need some changes in Python? (new PEP?)

**Comment by debonte** [comment, marked as answer]
This appears to work:

```python
T = TypeVar("T")

@overload
def get_value2(key_name: str, type_: Type[T]) -> T: ...

@overload
def get_value2(key_name: str) -> Any: ...

def get_value2(key_name: str, type_: Optional[Type] = None) -> Any: ...
```
